src/hipscat/catalog/pixel_node.py

```diff
@@ -0,0 +1,86 @@
+from __future__ import annotations
+
+from typing import List
+
+from hipscat.catalog.pixel_node_type import PixelNodeType
+
+
+class PixelNode:
+    """A node in the HIPSCat quadtree of HEALPix pixels
+
+    Attributes:
+        hp_order: HEALPix order of the pixel
+        hp_pixel: HEALPix pixel number
+        node_type: If the node is a leaf, root, or inner type
+        parent: The parent pixel node in the tree
+        children: The child nodes of the pixel
+
+    """
+
+    def __init__(self,
+                 hp_order: int | None,
+                 hp_pixel: int | None,
+                 node_type: PixelNodeType,
+                 parent: PixelNode | None = None,
+                 children: List[PixelNode] = None):
+        """Inits PixelNode with its attributes
+
+        Raises:
+            ValueError: Invalid arguments for the specified pixel type
+        """
+
+        if node_type == PixelNodeType.ROOT and parent is not None:
+            raise ValueError("Root node cannot have a parent")
+
+        if node_type == PixelNodeType.INNER or node_type == PixelNodeType.LEAF:
+            if parent is None:
```

Comment:
Is it worth checking here (or somewhere) that we're not introducing cycles? This should be a DAG (more specifically, a tree). Also, do we want to enforce `hp_order = parent.hp_order + 1`?

Reply:
I think enforcing that is a good idea. Would also mean root node has order -1 which I think makes sense. If we enforce this it should make sure there are no cycles.

I was thinking of validating the whole tree when the tree is created.